backend/graphql/types/residentType.ts

```diff
@@ -0,0 +1,78 @@
+import { gql } from "apollo-server-express";
+
+// TODO: Look into custom types for dates and date time types
+
+const residentType = gql`
+  type ResidentDTO {
+    userId: Int!
+    residentId: Int!
+    email: String!
+    phoneNumber: String
+    firstName: String!
+    lastName: String!
+    displayName: String
+    profilePictureURL: String
+    isActive: Boolean!
+    birthDate: Date!
+    roomNumber: Int!
+    credits: Float!
+    dateJoined: Date!
+    dateLeft: Date
+    notes: String
+  }
+
+  input CreateResidentDTO {
```

Comment:
Do we want all these fields to be required? For example, credits and dateJoined are auto generated in the database so it doesn't nessessarily need to be put in the createResidentApi, and potentially is info like roomNumber a  mandatory field?

Reply:
Good point on `credits` and `dateJoined`. We can make those optional. For now, we'll keep `roomNumber` as mandatory because we were informed by the clients that all residents are assigned a room number upon onboarding.

Reply:
I made the change just now @jeessh @ssreekar